src/System.Xml.XDocument/tests/xNodeBuilder/EndOfLineHandlingTests.cs

```diff
@@ -0,0 +1,310 @@
+            //[TestCase(Name = "NewLineHandling", Param = "XNodeBuilder")]
+            public partial class TCEOFHandling : BridgeHelpers
+            {
+                private XmlDiff _diff = null;
+                public TCEOFHandling()
+                {
+                    _diff = new XmlDiff();
+                }
+
+                private string ExpectedOutput(string input, NewLineHandling h, bool attr)
+                {
+                    string output = new string(input.ToCharArray());
+                    switch (h)
+                    {
+                        case NewLineHandling.Entitize:
+                            output = output.Replace("\r", "&#xD;");
+                            if (attr)
+                            {
+                                output = output.Replace("\n", "&#xA;");
+                                output = output.Replace("\t", "&#x9;");
+                            }
+                            break;
+                        case NewLineHandling.Replace:
+                            if (!attr)
+                            {
+                                output = output.Replace("\r\n", "\n");
+                                output = output.Replace("\r", "\n");
+                                output = output.Replace("\n", "\r\n");
+                            }
+                            else
+                            {
+                                output = output.Replace("\r", "&#xD;");
+                                output = output.Replace("\n", "&#xA;");
+                                output = output.Replace("\t", "&#x9;");
+                            }
+                            break;
+                        default:
+                            break;
+                    }
+                    return output;
+                }
+
+                //[Variation(Desc = "NewLineHandling Default value - NewLineHandling.Replace", Id = 1, Priority = 0)]
+                public void EOF_Handling_01()
+                {
+                    XmlWriterSettings wSettings = new XmlWriterSettings();
+                    TestLog.Compare(wSettings.NewLineHandling, NewLineHandling.Replace, "Incorect default value for XmlWriterSettings.NewLineHandling");
+                    XDocument d = new XDocument();
+                    XmlWriter w = CreateWriter(d);
+                    w.Dispose();
+                    TestLog.Compare(w.Settings.NewLineHandling, NewLineHandling.Replace, "Incorect default value for XmlWriter.Settings.NewLineHandling");
+                }
+
+                //[Variation(Desc = "XmlWriter creation with NewLineHandling.Entitize", Param = NewLineHandling.Entitize, Id = 2, Priority = 0)]
+                //[Variation(Desc = "XmlWriter creation with NewLineHandling.Replace", Param = NewLineHandling.Replace, Id = 3, Priority = 0)]
+                //[Variation(Desc = "XmlWriter creation with NewLineHandling.None", Param = NewLineHandling.None, Id = 4, Priority = 0)]
```

Comment:
Good question. In the past this library was doing something really similar to what Xunit's [Theory] is doing. At one point in time we converted the tests to not use it (and commented out the Variation attribute).

I decided not to remove this as it will make it easier to switch to [Theory]